### PR TITLE
Use Score for complexity calculation

### PR DESCRIPTION
**Pull request by briansheppard-toast**
Allow the parameters for endgame and middlegame complexity to vary independently.

This is identical to master in terms of calculations. The only difference will be the tradeoff between an mg_value() operation versus and addition. (Should be insignificant at most.)

The complexity calculation for midgame and endgame were already different, using "complexity + 50" for midgame, and "complexity" for endgame. The current change makes complexity into a Score, so that differences between midgame and endgame parameters are explicit. The use of Score adheres to Stockfish conventions in other areas of the code.

@Sopel97 commented: ... this patch provides additional parameters for potential tuning while having bar none impact on performance. It even comes with cleaner code.

@Tomtor replied: Having MORE parameters to tune is only an advantage if you CAN tune them and it results in an ELO gain. In the past we have removed parameters with the idea that it is an advantage that less parameters need to be tuned. The tuning capabilities in the fishtest framework are limited.

To summarize: Extra parameters might be valuable, and cleaner code (at least to my eyes).

No functional change.

LLR: 2.96 (-2.94,2.94) [-3.00,1.00]
Total: 34715 W: 7607 L: 7511 D: 19597

bench 4053577

### CODE REVIEW

**Comment by Rocky640** [comment]
My understanding of this initiative/complexity function is that it tries to measure what would happen if strongside uses his superiority to simplify (or threaten to simplify) into some king and pawn endgame. 

All the terms are king and pawn related only.
So this is essentially an endgame measure.
I would rather try to simplify the mg, then try to complexify it.

This PR might still be the basis of some interesting tuning experiments, but as it is,
I don't see why it should be committed into master

**Comment by Alayan-stk-2** [comment]
A parameter tuning based on this and gaining elo would be worth committing, but as-is, I agree with @Rocky640 ; there is no point.

**Comment by NKONSTANTAKIS** [comment]
Maybe let someone who is efficient with SPSA to tune it, leaving this pull open for some time as a reminder to focus on that direction as it looks promising.

**Comment by Vizvezdenec** [comment]
Tuning is good, why not. But committing "as is" would be strange. This is complification of current code.

**Comment by snicolet** [comment]
I am closing this now, for the reasons given in this thread. Let's us keep in mind the idea, however, if we manage to extract an Elo gain from the mg/eg tuning of complexity.
